### PR TITLE
User Profile: show site modal when site row tapped

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref #16245 

When the site is tapped on the User Profile, the site is displayed using the Reader site view it in a fullscreen modal.

The Reader site view is what is presented when tapping the site name on a Reader post card. The differences here are:
- The fullscreen modal.
- `X` left nav bar button to dismiss the view.

Otherwise they should look the same.
 
| Reader Site | User Site |
|--------|-------|
| reader_site | user_site |

---
To test:

Since we don't have real user info yet, I've stubbed in a bogus siteID. You'll need valid siteIDs for the modal to show. The [siteID](https://github.com/wordpress-mobile/WordPress-iOS/blob/e56b67b9d3d85419685980c57affc8f8fc8d6d13/WordPress/Classes/ViewRelated/User%20Profile%20Sheet/UserProfileSheetViewController.swift#L134) is set in  `UserProfileSheetViewController:showSite`.

---
With an invalid siteID (as-is):
- Enable the `newLikeNotifications` feature.
- Go to Notifications > Likes and tap a notification.
- Tap a user, then the site row.
- Nothing will happen visually. This error will appear in the logs: `Error: could not find ReaderSiteTopic for siteID [xxx]`

---
With a valid siteID:
- Set the siteID value as mentioned above.
- Go to Notifications > Likes and tap a notification.
- Tap a user, then the site row.
- Verify the site is displayed:
  - In a fullscreen modal.
  - With the `X` left nav bar button.

| iPhone | iPad |
|--------|-------|
| iphone | ipad |

---
## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is incomplete.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is incomplete.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature is incomplete.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Services/ReaderTopicService.h [hunk 1]
```diff
@@ -238,6 +238,16 @@ extern NSString * const ReaderTopicFreshlyPressedPathCommponent;
...
 
+/**
+ Find a site topic by siteID.
+ Combines `findSiteTopicWithSiteID` and `findSiteTopicWithFeedID` to get a topic by ID.
+ Useful when it's not known if the siteID is internal or external.
+ 
+ @param siteID The site id of the topic
+ @return A matched site topic
+ */
+- (ReaderSiteTopic *)siteTopicForSiteID:(NSNumber *)siteID;
```
Comment (aerych) [inline WordPress/Classes/Services/ReaderTopicService.h]: Being so similar to `findSiteTopicWithSiteID` I wonder if we could call it something like `siteTopicForSiteOrFeedID` to better distinguish it and make usage more obvious at a glance? Or even something like `siteTopicForUnknownID` or `siteTopicForGenericID`.

File: WordPress/Classes/Services/ReaderTopicService.m [hunk 2]
```diff
@@ -1176,6 +1176,11 @@ - (ReaderSiteTopic *)findSiteTopicWithFeedID:(NSNumber *)feedID
     return (ReaderSiteTopic *)[results firstObject];
 }
 
+- (ReaderSiteTopic *)siteTopicForSiteID:(NSNumber *)siteID
+{
+    return [self findSiteTopicWithSiteID:siteID] ?: [self findSiteTopicWithFeedID:siteID];
```
Comment (aerych) [inline WordPress/Classes/Services/ReaderTopicService.m]: These methods will return a topic that is already synced to the device, but it seems likely that the user won't already be following many likers and we'd need to fetch site or feed info in those cases. 
`siteTopicForSiteWithID:isFeed:success:failure:` is set up to do this. Maybe the existing method would be a better use case?  
If it helps, the siteIDs we get from the likes endpoints should never be feedIDs, just siteIDs. (I think. Pretty sure Jetpack sites are siteIDs.  FeedIDs would only be external sites crawled and stored in the feedbag).

**Review by aerych — CHANGES_REQUESTED** [review]
Howdy @ScoutHarris 👋 
Works as advertised if I happen to be following the topics I'm testing with, but I don't think we can count on that. I left some comments to that effect. Lemme know what you think.
Back to you!

**Comment by ScoutHarris** [comment]
Hey @aerych . 

Thank you for your help! I've made these changes:
- Per our Slack discussion, I've removed the logic to display external sites (and the poorly thought out `siteTopicForSiteID` method). I'll address external sites in a follow up. 
- The User Profile VC now uses `readerTopicService.siteTopicForSite` to fetch a site by ID. 

Ready for another review please.

**Review by aerych — APPROVED** [review]
Shiny! :shipit:  :)

**Pull request merged**